### DISCUSSION TITLE
Direct control over "managed" (geregeld)

### DISCUSSION THREAD

**Discussion by edegraaff** [emsesp/EMS-ESP32#1345] [General]
Is it possible to make use of ems-esp32 to direct switch on or off the  fireplace services or holiday service, today as sunday and tomorrow as sunday functions? 

Would be nice if i can switch them with home assistant directly 

regards

eelco

**Comment by proddy** [comment]
are those functions/features available on one of your EMS compatible devices?

**Reply by edegraaff** [reply]
Thes feature is availible in Nefit Easy and probably in several other Nefit devices i found a technical bulletin.
https://nefit-nl.resource.bosch.com/media/nefit/extranet_restricted/documentatie/technisch_bulletins/tb39.pdf
![image](https://github.com/emsesp/EMS-ESP32/assets/12442749/ceae05b2-7de8-4e72-93da-72dd8b0527ea)


Remeha seems to have a similar function

**Reply by proddy** [reply]
Ok. Then we need go hunting and find those telegrams. i will update the wiki to explain the best way to do this (https://emsesp.github.io/docs/Troubleshooting/#not-all-ems-devices-are-recognized-or-data-is-missing) but essentially, you need to turn on logging and then invoke the command on your thermostat so the telegrams are captured. Then we can mimick it in EMS-ESP.

**Reply by edegraaff** [reply]
Ok, when i have some guidance, i will be able to connect the pc to the esp and try to capture those telegrams. When i have it running i can try to pick all the special codes like sterilisation, shouwer timeblock, weeekend schedule, etc. I did some esp programming in the past, for HAM projects, i have a linux and a windows machine wit arduinosdk and as always .. putty :-)

**Reply by edegraaff** [reply]
Hi, i collected some logging with all logging on.   switched the fireplace function on and off several times. Attached the logging. 
[log.txt](https://github.com/emsesp/EMS-ESP32/files/13199051/log.txt)
Should i add a console device for more detailled logging? It is a bit hard to find the action of enabeling and disabeling the fireplace service in this log, maybe tou will notice it, if the esp does log the action it should be in it. 

If i need to take special actions to support, let me know, 

regards
eelco

**Reply by proddy** [reply]
thanks for the log. Can you also indicate the times when the fireplace was switch on and off?

**Reply by edegraaff** [reply]
if you look in the log 2023-10-29 18:32:56.278 and later are all related to the fireplace tests i have done, i have switched on and of for i think 4 times, would this help?

**Reply by proddy** [reply]
I'm still not certain what I'm suposed to looking for. How do you switch on/off the fireplace now?

**Reply by edegraaff** [reply]
I am sorry... I am now using the app that belongs to the themostat(Logamatic TC100/Moduline Easy), but i would like to integrate it into my home assistant. When this is possible i could add a sensor to the fireplace and activate a script to swith on the fireplace function of the 
GBx72/Trendline/Cerapur/Greenstar to bypass the thermostat, and let the Trendline, continue to heat the rest of the house. Does this make sense? 
Regards
Eelco

**Reply by proddy** [reply]
yes, makes sense. I just need you to turn on and off the fireplace with the official App while at the same time capturing some logs from EMS-ESP so I can see what commands (telegrams) are being sent to the Boiler. So timestamps are important since there will be a lot of traffic on the EMS bus. Then we can experiment adding these to EMS-ESP, first via the Custom Entity feature.

**Reply by edegraaff** [reply]
Ok, first i upgraded to the latest build, due the non responding issue. I will see if it solves the issue by monitoring with my NMS. 
Attached the logfile, i have switched on and off several times the fireplace function. I think the telegrams should be in this part of the logging. If you need additional logging from the console, with some instructions how to do this i will be able to deliver this also. 
2023-11-12 17:12:58.069 TRACE 107: [emsesp] boiler(0x08) -B-> All(0x00), UBAMonitorFast(0x18), data: 29 01 B8 4B 1C 09 01 25 40 01 73 80 00 01 70 00 ED 11 2D 48 00 C8 00 02 00
2023-11-12 17:12:58.271 TRACE 108: [emsesp] boiler(0x08) -B-> All(0x00), MC110Status(0x2A), data: 00 00 00 00 00 00 00 00 B0 00 B0 80 00 00 80 00 80 00 80 00 00
2023-11-12 17:12:58.550 TRACE 109: [emsesp] Me(0x0B) -R-> boiler(0x08), UBAParameters(0x16), length: 0x1B (offset 27)
2023-11-12 17:12:58.628 TRACE 111: [emsesp] boiler(0x08) -B-> All(0x00), UBAMonitorWW(0x34), data: 3E 01 73 80 00 21 00 00 01 00 01 36 AD 01 6D C3 00 80 00
2023-11-12 17:12:58.833 TRACE 112: [emsesp] boiler(0x08) -B-> All(0x00), UBADevices(0x07), data: 0B 00 01 00 00 00 00 00 00 00 00 00 00
2023-11-12 17:12:59.083 TRACE 113: [emsesp] Me(0x0B) -R-> boiler(0x08), UBAParameterWW(0x33), length: 0x1B
2023-11-12 17:12:59.122 TRACE 114: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWW(0x33), data: 08 FF 3C FB 00 28 00 00 46 D8 FF FF 00
2023-11-12 17:12:59.382 TRACE 115: [emsesp] Me(0x0B) -R-> boiler(0x08), UBASettingsWW(0x26), length: 0x1B
2023-11-12 17:12:59.407 TRACE 116: [emsesp] boiler(0x08) -W-> Me(0x0B), UBASettingsWW(0x26), data: 01 04 00 13 00 1E 5A 64
2023-11-12 17:12:59.581 TRACE 117: [emsesp] Me(0x0B) -R-> thermostat(0x18), EasyMonitor(0x0A), length: 0x1B
2023-11-12 17:12:59.701 TRACE 118: [emsesp] thermostat(0x18) -W-> Me(0x0B), EasyMonitor(0x0A), data: 00 00 00 00 00 00 00 00 07 44 07 3A 00 0C 3E CF 0E 88 00 00 00 0C 00 00 00 00
2023-11-12 17:13:00.016 TRACE 119: [emsesp] Me(0x0B) -R-> thermostat(0x18), EasyMonitor(0x0A), length: 0x1B (offset 26)
2023-11-12 17:13:08.096 TRACE 121: [emsesp] boiler(0x08) -B-> All(0x00), UBAMonitorFast(0x18), data: 29 01 B9 4B 1C 09 01 25 40 01 74 80 00 01 72 00 EC 11 2D 48 00 C8 00 02 00
2023-11-12 17:13:08.299 TRACE 122: [emsesp] boiler(0x08) -B-> All(0x00), MC110Status(0x2A), data: 00 00 00 00 00 00 00 00 B0 00 B0 80 00 00 80 00 80 00 80 00 00
2023-11-12 17:13:08.599 TRACE 123: [emsesp] boiler(0x08) -B-> All(0x00), UBAMonitorWW(0x34), data: 3E 01 74 80 00 21 00 00 01 00 01 36 AD 01 6D C3 00 80 00

The whole log after flashing the last firmware attached. 
[log (2).txt](https://github.com/emsesp/EMS-ESP32/files/13328551/log.2.txt)

I am curious ... 
Eelco

**Reply by edegraaff** [reply]
It seems when i switch on and off, the hermostat(0x18)  iw one of the telegrams that pass by. in this same set of data also the boiler(0x08) passes it could be those are not part of the fireplace action but i asume there will be an instruction to ignore thermostat and keep the cirquit operational with lets asume current water themperature, that way the system will heat to lets say a water temperature of 60C and keep the pump running or someting similar.

**Reply by proddy** [reply]
i can't see it quickly from the logs. Is there another way of switching it on/off from the thermostat or boiler directly and not using the app?

**Reply by edegraaff** [reply]
Hum, i could try this but i am not 100% sure it will work.  Thinking loud : 
Could we expect if telegrams will pass to the boiler, it should be visible, or could there be another route to manage the themostat? And or could it be ems-esp32 wont log the telegram because it does not recognise the data as telegram?  Would it be possible by using putty to the usb connector to tap into the esp, and get real time logging, with more detail? In that case i could sync clocks and change temperatures and see where data would pop up that is related to the fireplace overrule. Is it logic to expect the fireplace signal is similar to the others like, desinfection, or Holiday function, or today as sunday or tomorrow as sunday?  those functions are temporerely functions, the holiday function is an overrule over the normal clock program and says from this date to this date use this temp.  It should be possible to find the instruction somewhere in the datastreams...  i think ...  And for now:  all the best for 2024

**Reply by proddy** [reply]
for real-time, use the Console and the `log` command. See https://emsesp.github.io/docs/Console/#monitoring-the-ems-traffic

**Reply by MichaelDvP** [reply]
I think it's not possible, the holiday function is a thermostat function and the easy-thermostats are not writable and do not publish setting telegrams.

**Reply by edegraaff** [reply]
i will give it a try ans see what pops uo
